File: src/command/ConfigUpdate.ts
/*
 * Copyright 2021-2022 Golden_Water
 * SPDX-License-Identifier: AGPL-3.0-only
 */

import { Config } from "../utils/config/Config";

export type ConfigUpdateCmd = "updateConfig";

export type ConfigUpdate = {
  cmd: ConfigUpdateCmd;
  data: Config;
};

export function getConfigUpdateMessage(config: Config): ConfigUpdate {
  return {
    cmd: "updateConfig",
    data: config,
  };
}

export function getConfigUpdateCmd(): ConfigUpdateCmd {
  return "updateConfig";
}

File: share/src/command_packet/app_command/config_update.rs
/*
 * Copyright 2021-2023 Golden_Water
 * SPDX-License-Identifier: AGPL-3.0-only
 */

use crate::config::Config;

#[derive(serde::Serialize, serde::Deserialize, PartialEq, Debug, Clone)]
#[serde(rename_all = "camelCase")]
pub struct ConfigUpdate {
  config: Config,
}

impl ConfigUpdate {
  pub fn new(config: Config) -> ConfigUpdate {
    ConfigUpdate { config }
  }
}
